File: mobile/app/src/main/java/com/bellako/kiwi/userSettings/UserSettingsDTO.kt
package com.bellako.kiwi.userSettings

import com.bellako.kiwi.users.Email

data class UserSettingsDTO(
    val email: String,
    val soundVolume: Int,
    val musicVolume: Int
) {
    fun toState(): UserSettingsState {
        return UserSettingsState(
            email = email,
            soundVolume = soundVolume,
            musicVolume = musicVolume
        )
    }

    fun toDomainObject(): Result<UserSettings> {
        return Email.of(email).map { validEmail ->
            UserSettings(
                email = validEmail,
                soundVolume = soundVolume,
                musicVolume = musicVolume
            )
        }
    }
}

File: frontend/src/userSettings/UserSettingsDTO.ts
import {UserSettings} from "./UserSettings";
import {Email} from "../users/Email";

export interface UserSettingsDTO {
    email: string;
    soundVolume: number;
    musicVolume: number;
}

export const toDomainObject = (dto: UserSettingsDTO): UserSettings => ({
    email: Email.of(dto.email),
    soundVolume: dto.soundVolume,
    musicVolume: dto.musicVolume
});